cmd/clusterctl/client/cluster/topology.go

```diff
@@ -397,18 +399,27 @@ func (t *topologyClient) runDefaultAndValidationWebhooks(ctx context.Context, in
 		return errors.Wrap(err, "failed to run defaulting and validation on ClusterClasses")
 	}
 
-	// From the inputs gather all the objects that are not Clusters.
+	// From the inputs gather all the objects that are not Clusters or ClusterClasses.
 	// These objects will be used when initializing a dryrun client to use in the webhooks.
-	// We want to keep ClusterClasses in the webhook client. This is because validation of
-	// Cluster objects might need access to ClusterClass objects that are in the input.
 	filteredObjs = filterObjects(
 		in.Objs,
 		clusterv1.GroupVersion.WithKind("Cluster"),
+		clusterv1.GroupVersion.WithKind("ClusterClass"),
 	)
 	objs = []client.Object{}
 	for _, o := range filteredObjs {
 		objs = append(objs, o)
 	}
+	// Reconcile the ClusterClasses and add the reconciled version of them to the webhook client.
+	// This is required as validation of Cluster objects might need access to ClusterClass objects that are in the input.
+	// ClusterClass variable defaulting and validation relies on the ClusterClass `.status.variables` which is added
```

Comment:
```suggestion
	// Cluster variable defaulting and validation relies on the ClusterClass `.status.variables` which is added
```
Isn't it?

Reply:
>For ClusterClasses that are already in the Management Cluster I think it is fine for us to assume that they are already reconciled and we dont need to re-reconcile them.

Agreed, but I think it's useful to reconcile them anyway here as there might be useful reconciliation errors.

> If I understand this correctly, looks like the reconciledClusterClasses are not used when constructing the dry-run client in l.158. This would mean the Cluster reconciler will not operate on the reconciled CC and will exit early?

You're right - I added a variable with defaulting so this test now fails if the ClusterClass isn't also added to the reconciler. I also made sure the input ClusterClasses were replaced by the correctly reconciled versions.